### PR TITLE
Ian Hats!

### PR DESCRIPTION
**Pull request by ASTRALIXIX**
## About the PR
Adds support for 27 hats, 8 glasses and 2 helmets for the standard version of Ian!

## Why / Balance
SS13 has this! And God, these are hats for Ian!

## Technical details
Dog versions have been added to all rsi files with supported hats (So far only used by Ian). 
And a new "inventory_template" has been added

## Media
image

ian1

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
Nothing

**Changelog**
:cl:
- add: Added support for hats and glasses for Ian!
- tweak: Changed the sprite of Ian and Lisa.

### CODE REVIEW

**Comment by lzk228** [comment]
since most off them are only moved, i suggest to make a displacement map if it's possible

**Comment by UbaserB** [comment]
> New pet clothing - These need to use displacement maps. Gas tank sprites are still allowed.
Make sure to follow the freeze requirements in issue 8524

**Comment by ASTRALIXIX** [comment]
> > New pet clothing - These need to use displacement maps. Gas tank sprites are still allowed.
> 
> Make sure to follow the freeze requirements in issue 8524

I guess that changes the plans a bit, thanks.

**Comment by ASTRALIXIX** [comment]
> since most off them are only moved, i suggest to make a displacement map if it's possible

Theoretically, this is quite possible, but so far it is the only way

**Comment by ilovehans10** [comment]
Does Ian get the effects of wearing the functional "hats"? Can he wear sec glasses to see wanted status, or see people's health with a med-hud? Either way, this is a great feature!

**Review by TheShuEd — CHANGES_REQUESTED** [review]
Just use 1 displacement maps, dont add 100500 new sprites

**Comment by ASTRALIXIX** [comment]
> Just use 1 displacement maps, dont add 100500 new sprites

The problem is that the displacement map cannot support the sides of the sprites

File: Resources/Textures/Mobs/Pets/corgi.rsi/ian.png
Comment (Aeshus) [inline Resources/Textures/Mobs/Pets/corgi.rsi/ian.png:1]: I'm assuming the reason you offset the ian/lisa sprites themselves was to make the hats fit without a displacement map?

**Comment by Aeshus** [comment]
Also to confirm, these new sprites are from TGStation?

**Comment by TheShuEd** [comment]
> The problem is that the displacement map cannot support the sides of the sprites

Why not? It can

**Comment by ASTRALIXIX** [comment]
> Also to confirm, these new sprites are from TGStation?

Of course, maybe slightly modified

**Comment by ASTRALIXIX** [comment]
> > The problem is that the displacement map cannot support the sides of the sprites
> 
> Why not? It can

As far as I understand, displacement map cannot change sprite masks because it fully shifts the entire sprite

**Comment by Aeshus** [comment]
> > Also to confirm, these new sprites are from TGStation?
> 
> Of course, maybe slightly modified

Then you'll probably need to update the copyright modification in the `meta.json` to attribute whoever made modifications (you?).

File: Resources/Textures/Mobs/Pets/corgi.rsi/ian.png
Comment (ASTRALIXIX) [reply Resources/Textures/Mobs/Pets/corgi.rsi/ian.png:1]: Initially, yes, but later it was done just to simplify spriting

**Comment by ASTRALIXIX** [comment]
> Then you'll probably need to update the copyright information in the `meta.json` to attribute whoever made modifications (you?).

My changes were in a couple of pixels, I don't think it's critical. Still, I would try to implement it through displacement map.

**Comment by hyphenationc** [comment]
> > Then you'll probably need to update the copyright information in the `meta.json` to attribute whoever made modifications (you?).
> 
> My changes were in a couple of pixels, I don't think it's critical. Still, I would try to implement it through displacement map.

Please list attributions in full. This includes small changes, such as only a few pixels.

**Comment by TheShuEd** [comment]
> > > The problem is that the displacement map cannot support the sides of the sprites
> > 
> > 
> > Why not? It can
> 
> As far as I understand, displacement map cannot change sprite masks because it fully shifts the entire sprite. At least for the moment.

No, its offset field

Displacement maps can offset each pixel in different directions and different distance

**Comment by ASTRALIXIX** [comment]
> > > Why not? It can
> > 
> > 
> > As far as I understand, displacement map cannot change sprite masks because it fully shifts the entire sprite. At least for the moment.
> 
> No, its offset field
> 
> Displacement maps can offset each pixel in different directions and different distance

Oh, I get it, it changes everything a lot. Thanks

**Comment by TheShuEd** [comment]
https://docs.spacestation14.com/en/space-station-14/art/displacement-maps.html

**Comment by TheShuEd** [comment]
You can ping me in discord @eshhhed, if you have questions about displacement map workflow

**Pull request closed without merging**